x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecker.java

```diff
@@ -124,16 +137,33 @@ private DeprecationIssue ignoredOldIndicesCheck(
         IndexVersion currentCompatibilityVersion = indexMetadata.getCompatibilityVersion();
         // We intentionally exclude indices that are in data streams because they will be picked up by DataStreamDeprecationChecks
         if (DeprecatedIndexPredicate.reindexRequired(indexMetadata, true) && isNotDataStreamIndex(indexMetadata, clusterState)) {
-            return new DeprecationIssue(
-                DeprecationIssue.Level.WARNING,
-                "Old index with a compatibility version < 9.0 Has Been Ignored",
-                "https://www.elastic.co/guide/en/elasticsearch/reference/master/breaking-changes-9.0.html",
-                "This read-only index has version: "
-                    + currentCompatibilityVersion.toReleaseVersion()
-                    + " and will be supported as read-only in 9.0",
-                false,
-                meta(indexMetadata, indexToTransformIds)
-            );
+            var transforms = transformIdsForIndex(indexMetadata, indexToTransformIds);
+            if (transforms.isEmpty() == false) {
+                return new DeprecationIssue(
+                    DeprecationIssue.Level.WARNING,
+                    "One or more Transforms write to this old index with a compatibility version < 9.0",
+                    "https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-9.0.html"
+                        + "#breaking_90_transform_destination_index",
+                    Strings.format(
+                        "Transforms [%s] write to this index with version [%s] and cannot be supported as read-only in 9.0",
```

Comment:
I'm not sure a user would know what to do after reading this. It sounds a bit like if they upgrade to 9.0 their cluster might break because this incompatible index isn't ignored and "cannot be supported as read-only in 9.0"

Maybe something like:
> This read-only index was created in version 7.17 and will be supported as a read-only index in 9.0. The following transforms are no longer able to write to this index [%s].

Reply:
I agree to change the messaging based on @rudolf's input above. I'm not familiar with the subject, but would it be possible to add something actionable to the end of the message, something that the user can do to mitigate/solve the issue? For example:

```
This index was created in version 7.17 and will be supported as a read-only index in 9.0. The following transforms are no longer able to write to this index [%s]. Refer to the [migration guide](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-9.0.html) to learn more about how to handle your transforms destination indices."
```